hbase-server/src/main/java/org/apache/hadoop/hbase/master/snapshot/TakeSnapshotHandler.java

```diff
@@ -139,12 +140,17 @@ public TakeSnapshotHandler(SnapshotDescription snapshot, final MasterServices ma
   }
 
   private TableDescriptor loadTableDescriptor()
-      throws FileNotFoundException, IOException {
+      throws IOException {
     TableDescriptor htd =
       this.master.getTableDescriptors().get(snapshotTable);
     if (htd == null) {
       throw new IOException("TableDescriptor missing for " + snapshotTable);
     }
+    if(htd.getMaxFileSize()==-1 &&
```

Comment:
When we have max file size already set on a table (at src cluster) we will be preserving it in snapshot's HTD anyways right. Just confirming.
Also while taking snapshot, we can pass in this new config   hbase.snapshot.max.filesize.preserve = true right?
Very useful feature.

Reply:
> When we have max file size already set on a table (at src cluster) we will be preserving it in snapshot's HTD anyways right. Just confirming.

Yes, it won't be overridden by the system's region size configuration.

>Also while taking snapshot, we can pass in this new config hbase.snapshot.max.filesize.preserve = true right?
Very useful feature.

The current proposal sets this at the system level so all snapshots taken will use this. I'd also like this configuration to be passed at snapshot creation.

Reply:
Thanks for the reviews and suggestions @anoopsjohn @petersomogyi . I'm falling behind a bit, but will be putting it in an upcoming commit on the following days.

Reply:
>The current proposal sets this at the system level so all snapshots taken will use this. I'd also like this configuration to be passed at snapshot creation.
Sure. +1 for handling it as follow up issue